packages/ember-headless-form/src/components/headless-form.ts

```diff
@@ -34,6 +34,7 @@ export interface HeadlessFormComponentSignature<DATA extends UserData> {
   Element: HTMLFormElement;
   Args: {
     data?: DATA;
+    mutableData?: boolean;
```

Comment:
Wondering if `mutableData` may be confused with an object rather than a boolean on what it does if folks don't get autocomplete to help them?  Could maybe prefix with `is` or `enable` or switch to like `dataMode?: 'mutable' | 'immutable'`. 

Totally non-blocking, just thinking out loud!

Reply:
Naming is so incredibly hard!  It's the worst 😂. `dataMode` feels a bit better to me but maybe others can chime in later!

Reply:
`dataMode` sounds better if other data modes come up (i.e. not just mutable and immutable).

Reply:
I did switch to `@dataMode` now!